mzLib/MzLibUtil/PpmToleranceWithNotch.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MzLibUtil
+{
+    public class PpmToleranceWithNotch : Tolerance
+    {
+        private readonly double[] AcceptableSortedMassShifts;
+        public const double NotchStep = 1.00335483810;
+        public PpmTolerance PpmTolerance { get; init; }
+
+        public PpmToleranceWithNotch(double value, int protonNotches)
+            : base(value)
+        {
+            var massShifts = new List<double>();
+            for (int i = 1; i <= protonNotches; i++)
+            {
+                massShifts.Add(NotchStep * i);
+                massShifts.Add(-NotchStep * i);
+            }
+            AcceptableSortedMassShifts = massShifts.OrderBy(p => p).ToArray();
+            NumNotches = protonNotches;
+            PpmTolerance = new PpmTolerance(value);
+        }
+
+        int NumNotches { get; init; }
```

Comment:
Is this property necessary? If so it should be made public. If not, it should be removed